test/testutil/drbg.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "../testutil.h"
+#include <internal/rand.h>
+
+
+static int generate(void *vctx, unsigned char *out, size_t outlen,
+        unsigned int strength, int prediction_resistance,
+        const unsigned char *addin, size_t addin_len)
+{
+    for (; outlen != 0; --outlen)
+        *out++ = 1;
+    return 1;
+}
+
+EVP_RAND_CTX *test_setup_rand_swap(void)
+{
+    static EVP_RAND method;
```

Comment:
Great idea.  I didn't consider hooking this at the EVP_RAND layer.

Reply:
Thanks!  I'll finish hooking this up tomorrow, but let me know of any other comments you have.